packages/framework/presence/src/test/presenceManager.spec.ts

```diff
@@ -135,157 +195,199 @@ describe("Presence", () => {
 				});
 
 				describe("that is joining", () => {
-					let newAttendee: ISessionClient | undefined;
-
-					beforeEach(() => {
-						newAttendee = undefined;
-						afterCleanUp.push(
-							presence.events.on("attendeeJoined", (attendee) => {
-								assert(newAttendee === undefined, "Only one attendee should be announced");
-								newAttendee = attendee;
-							}),
-						);
-					});
-
-					function verifyNewAttendee(expectedConnectionId: ClientConnectionId): void {
-						assert(newAttendee !== undefined, "No attendee was announced");
-						assert.equal(
-							newAttendee.sessionId,
-							attendeeSessionId,
-							"Attendee has wrong session id",
-						);
-						assert.equal(
-							newAttendee.getConnectionId(),
-							expectedConnectionId,
-							"Attendee has wrong client connection id",
-						);
-						assert.equal(
-							newAttendee.getConnectionStatus(),
-							SessionClientStatus.Connected,
-							"Attendee connection status is not Connected",
-						);
-					}
-
 					it('first time is announced via `attendeeJoined` with status "Connected"', () => {
 						// Act - simulate join message from client
-						presence.processSignal("", initialAttendeeSignal, false);
+						const joinedAttendees = simulateAttendeeJoin([initialAttendeeSignalInfo]);
 
 						// Verify
-						verifyNewAttendee(initialAttendeeConnectionId);
+						assert(
+							joinedAttendees.length === 1 && joinedAttendees[0] !== undefined,
```

Comment:
`simulateAttendeeJoin` doesn't say it can return undefined. Why does undefined case need checked?

Reply:
Because of noUncheckedIndexedAccess, the compiler assumes that every value you get from indexing the array can be undefined

Reply:
noUncheckedIndexedAccess should be off - it is too agressive with ararys. We have a custom lint rule to take its place.

Reply:
Full enable of the replacement lint rule for presence is here: #23169 (please review :) )